File: php/features.php
<?php
declare(strict_types=1);

// Internetdb SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class InternetdbFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new InternetdbBaseFeature();
            case "test":
                return new InternetdbTestFeature();
            default:
                return new InternetdbBaseFeature();
        }
    }
}

File: rb/features.rb
# Internetdb SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module InternetdbFeatures
  def self.make_feature(name)
    case name
    when "base"
      InternetdbBaseFeature.new
    when "ratelimit"
      InternetdbRatelimitFeature.new
    when "retry"
      InternetdbRetryFeature.new
    when "test"
      InternetdbTestFeature.new
    when "timeout"
      InternetdbTimeoutFeature.new
    else
      InternetdbBaseFeature.new
    end
  end
end
